Common/cpp/SharedItems/ShareableValue.cpp

```diff
@@ -123,8 +129,11 @@ void ShareableValue::adapt(jsi::Runtime &rt, const jsi::Value &value, ValueType
       // create frozen object based on a copy of a given object
       type = ValueType::ObjectType;
       frozenObject = std::make_shared<FrozenObject>(rt, object, module);
+      containsHostFunction |= frozenObject->containsHostFunction;
```

Comment:
Should we add this check also for `MutableValue`? [here](https://github.com/software-mansion/react-native-reanimated/blob/c10217c5d6024f47af7911ec82c7d3c76fbd7fd0/Common/cpp/SharedItems/ShareableValue.cpp#L75)

For example for this situation: `const fn = makeMutable(() => {temp});`
Perhaps this is unusable edge case but I just ask

Reply:
I've thought about that but it seems like we do not apply that optimization in such a case, do we?

Reply:
You are right, we didn't.

Reply:
Thanks for the review. If you find anything like this bothering you you can always open a new PR/discussion, I'm merging this for now.